File: vm/tests/all/interopSingleBasic/lambdaExpr1.js
//JS

llvmFile = Polyglot.evalFile("llvm", "lambdaExpr1.so");

aObj = new llvmFile.AClass();
val = aObj.foo(4);
console.log("[JS] calling LLVM identity (x -> x) of 4 leads to: ", val);
aObj.foo = (x => 2*x-1);
val = aObj.foo(4);
console.log("[JS] calling LLVM (x -> 2x-1) of 4 leads to: ", val);

File: vm/tests/all/interopSingleBasic/lambdaExpr1.cpp
//CPP
#include<polyglot.h>

int identity(int x) {
	return x;
}

class AClass {
	public:
		int (*foo)(int);
		AClass();
};

POLYGLOT_DECLARE_CLASS(AClass);

AClass::AClass() {
	foo = &identity;
}

void* evaluate(void* aClass, void* argument) {
	AClass* aObj = polyglot_as_AClass(aClass);
	int64_t val = polyglot_as_int64(argument);
	int64_t returnValue = (*(aObj->foo))(val);
	void* returnObj = polyglot_from_int64(returnValue);
	free(aObj);
	return returnObj;
}
